gqcp/include/ONVBasis/SpinUnresolvedOperatorString.hpp

```diff
@@ -0,0 +1,107 @@
+
+#pragma once
+
+
+namespace GQCP {
+
+
+/**
+ *  A spin unresolved operator string.
+
+ *  An spin unresolved operator string represents a string of either annihilation or creation operators by its indices and a corresponding phase factor p.
+ *  For example, an operator string represented by indices <1, 2, 3> represents either:
+ *       p * a_1^\dagger a_2^\dagger a_3^\dagger
+ *  or
+ *      p * a_1 a_2 a_3.
+ *  
+ *  An operator string is always represented by the indices of the operators. Whether it denotes annihilation or creation operators depends on the context in which an operator string is used.
+ *  The operator strings are different from ONV's. Instead of representing the way orbitals are occupied, they purely represent the order of certain operators.
+ */
+class SpinUnresolvedOperatorString {
+private:
+    // The vector representing the indices of the annihilation or creation operators in the operator string.
+    std::vector<size_t> indices;
+
+    // The phase factor associated with this string of annihilation or creation operators.
+    int p;
+
+public:
+    /*
+     * MARK: Constructors
+     */
+
+    /**
+     *  Construct a `SpinUnresolvedOperatorString` from the vector of indices that it encapsulates.
+     * 
+     *  @param index_vector                The vector containing the operator indices.
+     */
+    SpinUnresolvedOperatorString(const std::vector<size_t>& index_vector) :
+        indices {index_vector},
+        p {1} {};
+
+
+    /**
+     *  Construct a `SpinUnresolvedOperatorString` from the vector of indices that it encapsulates and a specified phase factor.
+     * 
+     *  @param index_vector                The vector containing the operator indices.
+     *  @param phase_factor                The phase factor associated with the operator string, in case fermion anti-commutation rules have been used to modify the operator string. 
+     */
+    SpinUnresolvedOperatorString(const std::vector<size_t>& index_vector, int& phase_factor) :
+        indices {index_vector},
+        p {phase_factor} {};
+
+
+    /*
+     *  MARK: General information
+     */
+
+    /**
+     *  Retrieve the operator indices from the `SpinUnresolvedOperatorString`.
+     */
+    std::vector<size_t> operatorIndices() const { return this->indices; }
+
+    /**
+     *  Retrieve the phase factor corresponding to the `SpinUnresolvedOperatorString`.
+     */
+    int phaseFactor() const { return this->p; }
+
+    /**
+     * Check whether the operator string in question will result in zero when applied to the wave function.
+     *
+     * Note: There are different cases when an operator string will result in a zero value. This method checks all of them.
+     */
+    bool isZero() const {
+
+        // Check if the same index appears more than once in the operator string.
+        // Start by initializing a copy of the index vector associated with the operator string.
+        auto index_vector = this->operatorIndices();
```

Comment:
I didn't change this as the `sort` and `std::unique` functions change the vector and using `const auto& index_vector` would cause the compilation to fail.